lerobot/common/teleoperators/keyboard/teleop_keyboard.py

```diff
@@ -59,48 +58,53 @@ def __init__(self, config: KeyboardTeleopConfig):
         self.event_queue = Queue()
         self.current_pressed = {}
         self.listener = None
-        self.is_connected = False
+        self._is_connected = False
         self.logs = {}
 
     @property
     def action_feature(self) -> dict:
-        return {
-            "dtype": "float32",
-            "shape": (len(self.arm),),
-            "names": {"motors": list(self.arm.motors)},
-        }
+        # TODO(Steven): Change this when we agree what should this return
+        ...
 
     @property
     def feedback_feature(self) -> dict:
         return {}
 
+    @property
+    def is_connected(self) -> bool:
+        return self._is_connected
```

Comment:
I'd rather have something like this, wdyt?
```suggestion
        return isinstance(self.listener, keyboard.Listener) and self.listener.is_alive()
```

Reply:
While I agree we can improve the heuristic for checking connectivity, calling `is_connected` (or `connect`) without importing `pyinput` will runtime error with: `AttributeError: 'NoneType' object has no attribute 'Listener'`.
I suggest we make this more explicit and refine the connection check. If you're on board, I can open a separate PR against `robots_refactor` with an improvement proposal for `teleop_keyboard`, since it's independent of `lekiwi`.

Reply:
For my understanding: is there a case where we can use this class without having `pynput` imported?

Reply:
As it currently stands, even if `pynput` is unavailable (and thus not imported), an user can still create an instance of this class and access all its methods—including `connect()`, which would simply do nothing.

With the proposed change, instantiation would still be allowed, but calling `connect()` or `is_connected()` would raise: `AttributeError: 'NoneType' object has no attribute 'Listener'`.

IMO a better approach would be to raise a more proper error immediately in the constructor if `pynput` is missing. Since the class relies on keyboard functionality, there's no meaningful use for an instance without it.

Happy to implement whichever approach you think is best in this PR or in a new one - just let me know your preference!

Reply:
I just updated the suggestion so that there's no import issue

Reply:
Awesome, that would work too. I addressed the rest of the changes in this commit: https://github.com/huggingface/lerobot/pull/863/commits/e52bfc0795069c8235f3103695a1bc0f710b9338